catalog/core/catalog-core-urlresourcereader/src/main/java/ddf/catalog/resource/impl/URLResourceReader.java

```diff
@@ -370,6 +398,40 @@ private void skipBytes(InputStream is, String bytesToSkip) throws IOException {
             }
         }
     }
+        
+    private boolean validateFilePath(File resourceFilePath) throws IOException {
+        String resourceCanonicalPath = resourceFilePath.getCanonicalPath();
```

Comment:
Does canonical path remove stuff like /../../.. From the path so that something like /blah/../../../etc/passwd wouldn't match /blah and allow someone to download some file they shouldn't get to?